cylc/flow/task_pool.py

```diff
@@ -1419,7 +1417,10 @@ def spawn_on_output(self, itask, output, forced=False):
                     if not in_pool:
                         self.add_to_pool(t)
 
-                    if t.point <= self.runahead_limit_point:
+                    if (
+                        self.runahead_limit_point is not None
+                        and t.point <= self.runahead_limit_point
+                    ):
```

Comment:
Shouldn't this line have similar logic:

https://github.com/cylc/cylc-flow/blob/4b1adfc5b777102a8b9d65fdaa5ba1ce896b46a4/cylc/flow/task_pool.py#L781-L785

Given that `compute_runahead` can return `False` before updating the value:

https://github.com/cylc/cylc-flow/blob/4b1adfc5b777102a8b9d65fdaa5ba1ce896b46a4/cylc/flow/task_pool.py#L349-L350

Reply:
Are you saying add it to line 785? I.e.
```diff
**diff --git a/cylc/flow/task_pool.py b/cylc/flow/task_pool.py
index 01b0d03e91..c3574ee39a 100644
--- a/cylc/flow/task_pool.py
+++ b/cylc/flow/task_pool.py
@@ -784,4 +784,5 @@ class TaskPool:
 
         is_xtrig_sequential = False
+        if self.runahead_limit_point is not None:
             while point is not None and (point <= self.runahead_limit_point):
                 if tdef.is_parentless(point):
```

Reply:
No idea what the fix should be, but I think it's the same `runahead_limit_point` could be `None` problem you've fixed in the commented lines?

Reply:
Addressed